src/processor/queue-consumer.service.ts

```diff
@@ -164,38 +168,46 @@ export class QueueConsumerService extends WorkerHost implements OnApplicationBoo
     }
   }
 
+  private async setEpochCapacity(totalCapacityUsed: bigint): Promise<void> {
+    const capacityInfo = await this.blockchainService.capacityInfo(this.configService.getProviderId());
+    const currentEpoch = capacityInfo.currentEpoch;
+    const epochCapacityKey = `epochCapacity:${currentEpoch}`;
+    const epochCapacity = BigInt(await this.cacheManager.get(epochCapacityKey) ?? 0);
+    const newEpochCapacity = epochCapacity + totalCapacityUsed;
+    await this.cacheManager.set(epochCapacityKey, newEpochCapacity.toString());
+  }
+
   private async checkCapacity(): Promise<void> {
     const capacityLimit = this.configService.getCapacityLimit();
     const capacity = await this.blockchainService.capacityInfo(this.configService.getProviderId());
     const remainingCapacity = capacity.remainingCapacity;
-    let outOfCapacity = capacity.remainingCapacity <= 0n;
+    const currentEpoch = capacity.currentEpoch;
+    const epochCapacityKey = `epochCapacity:${currentEpoch}`;
+    const epochUsedCapacity = BigInt(await this.cacheManager.get(epochCapacityKey) ?? 0); // Fetch capacity used by the service
+    let outOfCapacity = remainingCapacity <= 0n;
 
-      if(!outOfCapacity) {
+    if(!outOfCapacity) {
         this.logger.debug(`Capacity remaining: ${remainingCapacity}`);
         if (capacityLimit.type === 'percentage') {
-          const minRemainingPct = 100 - capacityLimit.value;
-          const percentageRemaining = (capacity.remainingCapacity * 100n) / (capacity.totalCapacityIssued || 1n);
-          if (percentageRemaining <= minRemainingPct ) {
-            outOfCapacity = true;
-            this.logger.warn(
-              `Capacity threshold reached: remaining pct = ${percentageRemaining}% (${minRemainingPct}% required (${
-                (capacity.remainingCapacity * 100n) / capacity.totalCapacityIssued
-              }))`,
-            );
-          }
+            const capacityLimitPercentage = BigInt(capacityLimit.value);
+            const capacityLimitThreshold = (capacity.totalCapacityIssued * capacityLimitPercentage) / 100n;
+            this.logger.debug(`Capacity limit threshold: ${capacityLimitThreshold}`);
+            if (epochUsedCapacity >= capacityLimitThreshold) {
+                outOfCapacity = true;
+                this.logger.warn(`Capacity threshold reached: used ${epochUsedCapacity} of ${capacityLimitThreshold}`);
+            }
```

Comment:
if `epochCapacity` exceeds the % of total allowed then set `outOfCapacity` to be true